src/types/embedded.ts

```diff
@@ -0,0 +1,29 @@
+import deserialize from "../core/deserialize";
+import serialize from "../core/serialize";
+import { SKIP } from "../constants";
+import custom from "./custom";
+import { ClazzOrModelSchema } from "../api/types";
+
+/**
+ * This allows to embed the property values in the resulting json output
+ * and vice-versa.
+ *
+ * @param type {ClazzOrModelSchema<T>} Some class or model schema.
+ */
+export default function embedded<T>(type: ClazzOrModelSchema<T>) {
```

Comment:
Other built-in types all accept an ` additionalArgs?: AdditionalPropArgs` argument. We should do the same here.

For this type to be flexible, we probably need a "prefix" option.